src/acceptance-tests-brain/test-scripts/010_insecure_registry_test.rb

```diff
@@ -4,104 +4,155 @@
-    '/var/vcap/packages/docker-distribution/config.yml'
-FileUtils::Verbose.cp '/var/vcap/packages/acceptance-tests-brain/bin/docker-uploader',
-    '/var/vcap/packages/docker-distribution/bin/'
-FileUtils::Verbose.cp '/var/vcap/packages/acceptance-tests-brain/bin/registry',
-    '/var/vcap/packages/docker-distribution/bin/'
-at_exit do
-    set errxit: false do
-        run "cf delete -f secure-registry"
-        run "cf delete -f insecure-registry"
-        run "cf delete -f uploader"
+  run "cf create-shared-domain #{CF_TCP_DOMAIN} --router-group default-tcp"
+  run "cf update-quota default --reserved-route-ports -1"
+
+  puts "deploy ...................................................................."
+
+  # Deploy the registry ... Assemble the apps for push
+  FileUtils::Verbose.cp resource_path('docker-uploader/manifest.yml'),
+                        '/var/vcap/packages/docker-distribution/manifest.yml'
+  FileUtils::Verbose.cp resource_path('docker-uploader/config.yml'),
+                        '/var/vcap/packages/docker-distribution/config.yml'
+  FileUtils::Verbose.cp '/var/vcap/packages/acceptance-tests-brain/bin/docker-uploader',
+                        '/var/vcap/packages/docker-distribution/bin/'
+  FileUtils::Verbose.cp '/var/vcap/packages/acceptance-tests-brain/bin/registry',
+                        '/var/vcap/packages/docker-distribution/bin/'
+  at_exit do
+    set errexit: false do
+      puts "........................................................................... SHUTDOWN"
+      run "cf delete -f secure-registry"
+      run "cf delete -f insecure-registry"
+      run "cf delete -f uploader"
     end
-end
-run "cf push -f manifest.yml --var tcp-domain=#{CF_TCP_DOMAIN}",
-    chdir: '/var/vcap/packages/docker-distribution/'
+  end
+  run "cf push -f manifest.yml --var tcp-domain=#{CF_TCP_DOMAIN}",
+      chdir: '/var/vcap/packages/docker-distribution/'
 
-run 'cf apps'
+  run 'cf apps'
 
-REGISTRIES.each_pair do |regname, registry_url|
+  # With the apps running, now watch (tail) all their logs. (Dumped into helper files)
+  # Whenever needed we simply cat the relevant logs into the test output.
+
+  threads << Thread.new{
+    run "cf logs uploader > #{tmpdir}/app-uploader.log"
+  }
+
+  REGISTRIES.each_pair do |regname, registry_url|
+    threads << Thread.new{
+      run "cf logs #{regname} > #{tmpdir}/app-#{regname}.log"
+    }
+  end
+
+  threads.each { |th| th.abort_on_exception = true }
+
+  # Wait a bit to have the log tailer thread start properly and settle before doing more.
+  sleep 5
+
+  REGISTRIES.each_pair do |regname, registry_url|
+    puts "wait for registry to be available ......................................... #{regname}"
     # Wait for the registry to be available
     run_with_retry 60, 1 do
-        run "curl -kv #{registry_url}/v2/"
+      run "curl -kv #{registry_url}/v2/"
     end
+  end
+  run_with_retry 60, 1 do
+    run "curl -kv http://uploader.#{ENV['CF_DOMAIN']}"
```

Comment:
This doesn't have `--fail`; does that actually wait until it's up?

Reply:
Added `--fail` in commit b86a1aea256d52838f3584c756a75dfe0b8147c5.
From the quick look at the manpage this does seem to only suppress output and prevent conversion of errors, not be needed to have errors at all. To wit

```
Fail silently (no output at all) on server errors. This is mostly done to better enable scripts etc to better deal with failed attempts. In normal cases when an HTTP server fails to deliver a document, it returns an HTML document stating so (which often also describes why and more). This flag will prevent curl from outputting that and return error 22.
```